liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -102,17 +109,28 @@ public void onConnectionSuspended(int cause) {
   public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
   }
 
+  @SuppressLint("MissingPermission")
   @Override
   public Location getLastLocation() {
```

Comment:
I would suggest deprecating `getLastKnownLocation()` and  introducing a paradigm similar to Google by using success and failure listener so we have an api as `getLastKnowLocation(SomeSuccessCallback<Location>, SomeFailCallback<Error>)`. Introducing this change will also allow us to be more flexible in other implementations of LocationEngine.  if someone would require performing an async operation as part of this API, they would be able to vs null value returns.